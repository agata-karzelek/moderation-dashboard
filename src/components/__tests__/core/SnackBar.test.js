import React from "react";
import { shallow } from "enzyme";
import SnackBar from "../../core/SnackBar";

describe("Test Cases for SnackBar", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<SnackBar status="info" msg="No applications found" />);
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("check handleClose", () => {
    expect(wrapper.find("#snackBar").getElement().props.open).toBeTruthy();

    wrapper.find("#snackBar").getElement().props.onClose("", "clickaway");
    expect(wrapper.find("#snackBar").getElement().props.open).toBeTruthy();

    wrapper.find("#snackBar").getElement().props.onClose("", "");
    expect(wrapper.find("#snackBar").getElement().props.open).toBeFalsy();
  });
});
