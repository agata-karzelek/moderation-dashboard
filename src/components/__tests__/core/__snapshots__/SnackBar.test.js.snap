// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cases for SnackBar Snapshot 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  autoHideDuration={9000}
  id="snackBar"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(Alert))
    onClose={[Function]}
    severity="info"
  >
    No applications found
  </WithStyles(ForwardRef(Alert))>
</WithStyles(ForwardRef(Snackbar))>
`;
